src/ticks.js

```diff
@@ -2,35 +2,42 @@ const e10 = Math.sqrt(50),
     e5 = Math.sqrt(10),
     e2 = Math.sqrt(2);
 
-function tickSpec(start, stop, count) {
+const logParams = {
+  binary: [Math.log2, 2],
+  natural: [(n) => Math.log1p(n - 1), Math.E],
+  common: [Math.log10, 10],
+}
+
+function tickSpec(start, stop, count, log = 'common') {
+  const [logFn, base] = logParams[log] || logParams.common;
   const step = (stop - start) / Math.max(0, count),
-      power = Math.floor(Math.log10(step)),
-      error = step / Math.pow(10, power),
-      factor = error >= e10 ? 10 : error >= e5 ? 5 : error >= e2 ? 2 : 1;
+    power = Math.floor(logFn(step)),
+    error = step / Math.pow(base, power),
+    factor = error >= e10 ? 10 : error >= e5 ? 5 : error >= e2 ? 2 : 1;
   let i1, i2, inc;
   if (power < 0) {
-    inc = Math.pow(10, -power) / factor;
+    inc = Math.pow(base, -power) / factor;
     i1 = Math.round(start * inc);
     i2 = Math.round(stop * inc);
     if (i1 / inc < start) ++i1;
     if (i2 / inc > stop) --i2;
     inc = -inc;
   } else {
-    inc = Math.pow(10, power) * factor;
+    inc = Math.pow(base, power) * factor;
     i1 = Math.round(start / inc);
     i2 = Math.round(stop / inc);
     if (i1 * inc < start) ++i1;
     if (i2 * inc > stop) --i2;
   }
-  if (i2 < i1 && 0.5 <= count && count < 2) return tickSpec(start, stop, count * 2);
+  if (i2 < i1 && 0.5 <= count && count < 2) return tickSpec(start, stop, count * 2, log);
   return [i1, i2, inc];
 }
 
-export default function ticks(start, stop, count) {
+export default function ticks(start, stop, count, log) {
```

Comment:
> Too many positional arguments (start, stop, count, base). Overall, I think I would prefer to keep the existing methods as-is and have base-specific variants of d3.ticks and d3.tickIncrement for base = 2 (binary) and base = e (“natural”).

I'm happy to separate these added methods into a separate methods. But I'd like to avoid duplication of code. Would you be ok with having the `log` option on a base function then exporting these separate log type methods with just `start`, `stop` and `count` args?

So exporting `ticks`, `ticksNatural` and `ticksBinary`, along with their respective `tickIncrement` functions.